src/SystemWebAdapters/src/Middleware/BufferRequestStreamMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using System.Web.Metadata;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+
+namespace System.Web.Middleware;
+
+internal class BufferRequestStreamMiddleware : IMiddleware
+{
+    private readonly ILogger<BufferRequestStreamMiddleware> _logger;
+
+    public BufferRequestStreamMiddleware(ILogger<BufferRequestStreamMiddleware> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContextCore context, RequestDelegate next)
+    {
+        var metadata = context.GetEndpoint()?.Metadata.GetMetadata<IBufferRequestStreamMetadata>();
+
+        if (metadata is not null && metadata.IsEnabled)
```

Comment:
C# 10 is calling you...

```suggestion
        if (metadata is { IsEnabled: true })
```